sqlmesh/core/model/definition.py

```diff
@@ -553,16 +553,17 @@ def update_schema(
                     {k: v.sql(dialect=self.dialect) for k, v in mapping_schema.items()},
                 )
             else:
+                logger.warning(
```

Comment:
Wouldn't this result in warning for models that are on the edge (at the root for DAG) even if they carefully list all projections and their types? Not sure this is a good experience.

Reply:
i think maybe we should go with the approach where we don't clear the mapping schema

Reply:
yea we were just chatting about this, i'll look into refactoring that section to have a more explicit signal for avoiding the optimizer